quarkc/compiler.py

```diff
@@ -350,6 +350,10 @@ def supertypes(self):
         for sup in self.supertypes(self.type):
             yield sup
 
+    @overload(AST)
```

Comment:
What's "supertypes" do? And why does it sometimes take an argument and sometimes not? (Some docstrings might help).